src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -423,6 +492,30 @@ public virtual Publisher.IPublisherClient GrpcClient
 
     }
 
+    // Something like this will be in GAX
+    public class ApiCall<TRequest, TResponse>
+        where TRequest : class, IMessage<TRequest>
+        where TResponse : class, IMessage<TResponse>
+    {
+        // May or may not keep these delegates. Using Func<>s instead is a little ugly.
```

Comment:
It is a bit more boilerplat-y but it is not ugly per se... Now if there were typedefs in C# you could fix that easily...